ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -369,87 +373,162 @@ func (scbuilder *SiteConfigBuilder) getWorkloadManifest(cpuSet string, role stri
 	return workloadFileForRole, reflect.ValueOf(workloadStr).Interface(), nil
 }
 
-func (scbuilder *SiteConfigBuilder) getExtraManifest(dataMap map[string]interface{}, clusterSpec Clusters) (map[string]interface{}, error) {
-	// Figure out the list of node roles we need to support in this cluster
-	roles := map[string]bool{}
-	for _, node := range clusterSpec.Nodes {
-		roles[node.Role] = true
-	}
+// getExtraManifestMaps return 2 maps and error
+// first map consists: key as fileName, value as file content
+// second map contains: key as machineConfigs filename, value as true/false
+func (scbuilder *SiteConfigBuilder) getExtraManifestMaps(filePath interface{}, roles map[string]bool, clusterSpec Clusters) (map[string]interface{}, map[string]bool, error) {
 
-	// Adding the pre-defined DU profile extra-manifest.
-	files, err := GetExtraManifestResourceFiles(scbuilder.scBuilderExtraManifestPath)
-	if err != nil {
-		return nil, err
-	}
+	var (
+		files []fs.FileInfo
+		err   error
+	)
 
+	dirFiles := &DirContainFiles{}
+	dirFilesArray := []DirContainFiles{}
+	dataMap := make(map[string]interface{})
 	// Manifests to be excluded from merging
 	doNotMerge := make(map[string]bool)
 
-	for _, file := range files {
-		if file.IsDir() || file.Name()[0] == '.' {
-			continue
+	// do a type check
+	switch path := filePath.(type) {
```

Comment:
better to have the filePath given as []string type to the function and change the code accordingly. Otherwise you need to check for any other type and return error.

Reply:
Actually the filepath only can be either string (i.e. scBuilderExtraManifestPath) or []string (ExtraManifests.SearchPaths), no other type is possible here. Also, when the filePath is string we are querying the file path using `GetExtraManifestResourceFiles()`, which build the filePaths targeting where executable is. On the other hand, for []string case we are calling `GetFiles()` which looking for files at current directory.

So, based on your feedback, i will add a type check and return error.

Reply:
case widened and added unit test for type check in commit 4e71357842c233ccb388973c901535f4c82919cd

Reply:
so if the type will be only string or []string better to keep the func parameter clean like that
`func (scbuilder *SiteConfigBuilder) getExtraManifestMaps(filePath ...string, roles map[string]bool, clusterSpec Clusters)`
this way you can pass string or []string

Reply:
but `filePath ...string` doesn't accept pointer to slice *[]string